.github/workflows/release.yml

```diff
@@ -124,7 +131,6 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           CGO_ENABLED: 1
-          GO111MODULE: off
```

Comment:
IIRC this was added to build gw in GOPATH mode, so that plugins built by plugin compiler would work without issues.

Reply:
Was removed to resolve an issue. We have a few weeks of testing to see if we need to update CI with new tests here, which currently all pass...

Reply:
Yes, this can be very sensitive to Go plugins. Smoke tests which run plugins tests are running only on refs, so thats why this did not catched it.

Reply:
The changes here are mainly related to go tooling. Cmds like go install, and go list, should ignore GO111MODULE (use: auto) and not modify go.mod; however,

```
  ⨯ release failed after 0s                  error=failed to get module path: exit status 1: go: list -m cannot be used with GO111MODULE=off
```

This was introduced/announced in https://go.dev/blog/go116-module-changes. Effectively this becomes GO111MODULE=auto, and our plugin builds are vendored and by docs, vendored builds do not use go modules. Release pipeline needs some love maybe, seems like the issue should trigger a goreleaser version check, maybe it needs updating to ignore the env settings for the go toolchain invocations where appropriate.

cc @alephnull

Reply:
The [latest goreleaser](https://github.com/TykTechnologies/tyk/blob/master/.github/workflows/release.yml#L125) is used in the release pipelines.